packages/sources/booster.move

```diff
@@ -0,0 +1,154 @@
+module refund::booster {
+    use sui::tx_context::{TxContext, sender};
+	use sui::coin::{Self, Coin};
+    use sui::transfer;
+    use sui::balance;
+    use sui::table;
+    use sui::dynamic_field as df;
+    use sui::object::{Self, UID};
+    use sui::package::Publisher;
+    use sui::sui::SUI;
+    
+    use refund::refund::{
+        Self, RefundPool, claim_refund_,
+        accounting, unclaimed, uid_mut,
+        accounting_mut, booster_pool_mut
+    };
+    use refund::accounting::{
+        total_raised_for_boost, total_raised_for_boost_mut,
+        current_liabilities, total_boosted_mut, total_unclaimed_boosted
+    };
+    use refund::math::div;
+    use refund::table::{Self as refund_table};
+    use refund::pool::{funders_mut, funds, funds_mut};
+
+    const EAddressRetrievedBoostCap: u64 = 0;
+    const EInsufficientFunds: u64 = 1;
+
+    struct BoostedCapDfKey has copy, store, drop { affected_address: address }
+
+    struct BoostedClaimCap has key {
+        id: UID,
+        new_address: address
+    }
+
+    // === Phase 2: Funding ===
+
+    public entry fun fund(
+        pool: &mut RefundPool,
+        coin: Coin<SUI>,
+        ctx: &mut TxContext,
+    ) {
+        refund::assert_funding_phase(pool);
+        
+        let amount = coin::value(&coin);
+        let total_raised_for_boost = total_raised_for_boost_mut(accounting_mut(pool));
+        *total_raised_for_boost = *total_raised_for_boost + amount;
+
+        let booster = booster_pool_mut(pool);
+        refund_table::insert_or_add(funders_mut(booster), sender(ctx), amount);
+        balance::join(funds_mut(booster), coin::into_balance(coin)); 
+    }
+
+    // === Phase 3: Claim Refund ===
+
+    // called via tx sent from Rinbot address
+    public fun allow_boosted_claim(
+        pub: &Publisher,
+        pool: &mut RefundPool,
+        affected_address: address,
+        new_address: address,
+        ctx: &mut TxContext
+    ) {
+        refund::assert_publisher(pub);
+        refund::assert_claim_phase(pool);
+        refund::assert_address(pool, affected_address);
+        assert!(!df::exists_(uid_mut(pool), BoostedCapDfKey { affected_address }), EAddressRetrievedBoostCap);
+
+        let boosted_claim_cap = BoostedClaimCap {
+            id: object::new(ctx),
+            new_address,
+        };
+
+        df::add(uid_mut(pool), BoostedCapDfKey { affected_address }, true);
+        transfer::transfer(boosted_claim_cap, affected_address);
+    }
+
+    public entry fun claim_refund_boosted(
```

Comment:
Sorry for reposting this question, it's still a bit not clear to me.

In the "boosted" claims, you, as the holder of "publisher", can strictly define the "new_address" for an "affected_address", and this function is callable from that affected_address (as they'll receive the cap), but in the end of the day, all the funds will be transferred into the "new_address" that is saved. 

What is this `new_address` in that function, and why do you (`Publisher` owner) control it?

Reply:
The "boosted" claims will be allowed only if the user allocates the funds for buying and selling tokens or NFTs. We use a backend wallet associated to the user for this in our telegram trading bot, which allows users to withdraw profits, however the 150% notional remains the in the account without the user being able to transfer the funds out the backend wallet associated to the user.

The `new_address` therefore represents the user's newly generated backend wallet on the trading bot. Therefore the flow is the following:

1. We generate a backend wallet for the user and link it to the user's affected wallet by calling the permissioned endpoint `allow_boosted_claim`;
2. The user then calls `claim_refund_boosted` by signing with their affected wallet, therefore transferring the entirety of the funds to their backend wallet

The reason why `allow_boosted_claim` is permissioned is to make sure the funds are ultimately transferred to the backend wallet and the reason for the seconds step `claim_refund_boosted` requiring the affected wallet signature is to make sure the flow ultimately is consented by the user.

I hope this answers is :)

Reply:
Thanks for elaborating!